docs/tutorials/configure/pet-photographer.md

```diff
@@ -0,0 +1,840 @@
+The camera file is at [`rdk/components/camera/client.go`](https://github.com/viamrobotics/rdk/components/camera/client.go).
+
+You can create your own code or copy the code for your file [color_filter.py](https://github.com/viam-labs/modular-filter-examples/blob/main/pycolorfilter/color_filter.py) and [color_filter.go](https://github.com/viam-labs/modular-filter-examples/blob/main/colorfilter/color_filter.go).
+
+For more information, refer to [Create a custom module](https://docs.viam.com/modular-resources/create/#create-a-custom-module).
+
+#### Include the filter module requirements
+
+Your custom filter module must contain two critical elements in your filter function.
+One will check if the caller of the filter function is the data management service.
+The other will ensure that if the data management service is not the caller, an error is returned.
+
+#### Check the caller of the collector function
+
+When creating your own filter module, it's required to check whether the data management service is the caller of the function responsible for data capture to prevent unwanted effects on the filter state.
+
+You can achieve this by examining the `fromDataManagement` value within the `extra` argument passed to your filter function in your <file>color_filter.py</file> or <file>color_filter.go</file> file.
+
+The approach for checking this varies depending on the programming language used to configure your camera:
+
+{{< tabs >}}
+{{% tab name="Python"%}}
+First, import `from_dm_from_extra`:
+
+```python {class="line-numbers linkable-line-numbers"}
+from viam.utils import from_dm_from_extra
+```
+
+Then, include it in the conditional statement in your filter function:
+
+```python {class="line-numbers linkable-line-numbers"}
+     if from_dm_from_extra(extra):
+         detections = await self.vision_service.get_detections(img)
+```
+
+- If `from_dm_from_extra` is `true` and the data management service is the caller of your filter function, the vision service is called to obtain an image and store it within the `img` variable.
+
+{{% /tab %}}
+{{% tab name="Go"%}}
+
+Write a conditional statement that checks `data.FromDMContextKey{}`:
+
+```go {class="line-numbers linkable-line-numbers"}
+if ctx.Value(data.FromDMContextKey{}) != true {
+   // If not data management collector, return underlying stream contents without filtering.
+   return fs.cameraStream.Next(ctx)
+}
+
+// Only return captured image if it contains a certain color set by the vision service.
+ img, release, err := fs.cameraStream.Next(ctx)
+
+ detections, err := fs.visionService.Detections(ctx, img, map[string]interface{}{})
+```
+
+- If `data.FromDMContextKey{}` is `true` and the data management service is the caller, captures an image by declaring the (`img`) variable and filling it with the content from the camera stream.
+- Then, after capturing the image, the code continues to request detections.
+
+**PLACEHOLDER: `FromDMContextKey` vs `FromDMString`?**
```

Comment:
That's a question for @kaywux right?

Reply:
Yes, when she's back I'll be asking her about it!

Reply:
The below is somewhat in reverse: FromDMContextKey is _only_ used for camera, and all other components (including sensor) using FromDMString in the Go SDK